packages/medusa/src/api/routes/admin/products/transaction/create-product-variant.ts

```diff
@@ -78,52 +78,76 @@ export const createVariantTransaction = async (
+  async function createVariants(variantInput: CreateProductVariantInput[]) {
     return await productVariantServiceTx.create(productId, variantInput)
   }
 
-  async function removeVariant(variant: ProductVariant) {
-    if (variant) {
-      await productVariantServiceTx.delete(variant.id)
+  async function removeVariants(variants: ProductVariant[]) {
+    if (variants.length) {
+      await productVariantServiceTx.delete(variants.map((v) => v.id))
     }
   }
 
-  async function createInventoryItem(variant: ProductVariant) {
-    if (!variant.manage_inventory) {
-      return
+  async function createInventoryItems(variants: ProductVariant[] = []) {
+    const context = { transactionManager: manager }
+
+    let results: InventoryItemDTO[] = []
+
+    for (const variant of variants) {
+      if (!variant.manage_inventory) {
+        continue
+      }
+
+      const result = await inventoryService!.createInventoryItem(
+        {
+          sku: variant.sku,
+          origin_country: variant.origin_country,
+          hs_code: variant.hs_code,
+          mid_code: variant.mid_code,
+          material: variant.material,
+          weight: variant.weight,
+          length: variant.length,
+          height: variant.height,
+          width: variant.width,
+        },
+        context
+      )
+
+      results.push(result)
     }
 
-    return await inventoryService!.createInventoryItem({
-      sku: variant.sku,
-      origin_country: variant.origin_country,
-      hs_code: variant.hs_code,
-      mid_code: variant.mid_code,
-      material: variant.material,
-      weight: variant.weight,
-      length: variant.length,
-      height: variant.height,
-      width: variant.width,
-    })
+    return results
   }
 
-  async function removeInventoryItem(inventoryItem: InventoryItemDTO) {
-    if (inventoryItem) {
-      await inventoryService!.deleteInventoryItem(inventoryItem.id)
-    }
+  async function removeInventoryItems(inventoryItems: InventoryItemDTO[] = []) {
+    const context = { transactionManager: manager }
+
+    return await Promise.all(
+      inventoryItems.map(async (inventoryItem) => {
+        return await inventoryService!.deleteInventoryItem(
+          inventoryItem.id,
+          context
+        )
+      })
+    )
   }
 
-  async function attachInventoryItem(
-    variant: ProductVariant,
-    inventoryItem: InventoryItemDTO
+  async function attachInventoryItems(
+    variants: ProductVariant[],
+    inventoryItems: InventoryItemDTO[]
   ) {
-    if (!variant.manage_inventory) {
-      return
-    }
+    for (const variant of variants) {
+      if (!variant.manage_inventory) {
+        continue
+      }
 
-    await productVariantInventoryServiceTx.attachInventoryItem(
-      variant.id,
-      inventoryItem.id
-    )
+      const inventoryItem = inventoryItems.find((i) => i.sku === variant.sku)!
+
+      await productVariantInventoryServiceTx.attachInventoryItem(
+        variant.id,
+        inventoryItem.id
```

Comment:
suggestion(nit): make a `attachInventoryItems` method to take a list of `{variant_id: string, inventory_item_id: string}` to insert. 

note: not as important as the one above since this is internal to the core

Reply:
Same as above, but if you agree I can do it

Reply:
It is just to avoid expanding the scope of the pr too much

Reply:
Done